src/main/java/io/harness/cf/client/api/MetricsProcessor.java

```diff
@@ -117,29 +129,27 @@ public void pushToQueue(Target target, String featureName, Variation variation)
 
   void registerEvaluation(Target target, String featureName, Variation variation) {
 
-    final int freqMapSize = frequencyMap.size();
-
-    if (freqMapSize > config.getBufferSize()) {
-      if (log.isWarnEnabled()) {
-        log.warn(
-            "Metric frequency map exceeded buffer size ({} > {}), force flushing",
-            freqMapSize,
-            config.getBufferSize());
-      }
-      // If the map is starting to grow too much then push the events now and reset the counters
-      scheduler.schedule(this::runOneIteration, 0, SECONDS);
-    }
-
     Target metricTarget = globalTarget;
 
     if (target != null) {
-      uniqueTargetSet.add(target);
-      if (!config.isGlobalTargetEnabled()) {
-        metricTarget = target;
+
+      if (!targetsSeen.contains(target) && targetsSeen.size() + 1 > MAX_SENT_TARGETS_TO_RETAIN) {
+        log.warn("Target count has exceeded {}", MAX_SENT_TARGETS_TO_RETAIN);
```

Comment:
I think we should only log this warning once per metrics interval. It could get quite spammy otherwise. Fwiw the change we made in .NET logs it once, so might be good to align.

Reply:
I'm in the process of making this change, as the unit tests are showing it spam